ui/pages/confirmations/components/simulation-details/useBalanceChanges.ts

```diff
@@ -106,7 +106,7 @@ function getNativeBalanceChange(
   }
   const asset = NATIVE_ASSET_IDENTIFIER;
   const amount = getAssetAmount(nativeBalanceChange, NATIVE_DECIMALS);
-  const fiatAmount = amount.times(nativeFiatRate).toNumber();
+  const fiatAmount = amount?.times(nativeFiatRate).toNumber();
```

Comment:
As below, do we need to ask why the amount is `null` or `undefined`?

`getAssetAmount` just appears to primarily parse the value but we return `undefined` above if it was missing.

Does this suggest the `difference` value in the `simulationData` is bad?

Reply:
Looking at the return type of getAssetAmount, BigNumber, this should never be undefined.